spec/files/reporting_measure_raise/measures/purposefully_broken_reporting_measure/measure.rb

```diff
@@ -0,0 +1,163 @@
+
+    request = OpenStudio::IdfObject.load('Output:Variable,,Site Outdoor Air Drybulb Temperature,Hourly;').get
+    result << request
+
+    return result
+  end
+
+  # define what happens when the measure is run
+  def run(runner, user_arguments)
+    super(runner, user_arguments)
+
+    # use the built-in error checking
+    if !runner.validateUserArguments(arguments, user_arguments)
+      return false
+    end
+
+    # get the last model and sql file
+    model = runner.lastOpenStudioModel
+    if model.empty?
+      runner.registerError('Cannot find last model.')
+      return false
+    end
+    model = model.get
+
+    sql_file = runner.lastEnergyPlusSqlFile
+    if sql_file.empty?
+      runner.registerError('Cannot find last sql file.')
+      return false
+    end
+    sql_file = sql_file.get
+    model.setSqlFile(sql_file)
+
+    # put data into the local variable 'output', all local variables are available for erb to use when configuring the input html file
+
+    output =  'Measure Name = ' << name << '<br>'
+    output << 'Building Name = ' << model.getBuilding.name.get << '<br>' # optional variable
+    output << 'Floor Area = ' << model.getBuilding.floorArea.to_s << '<br>' # double variable
+    output << 'Floor to Floor Height = ' << model.getBuilding.nominalFloortoFloorHeight.to_s << ' (m)<br>' # double variable
+    output << 'Net Site Energy = ' << sql_file.netSiteEnergy.to_s << ' (GJ)<br>' # double variable
+
+    # read in template
+    html_in_path = "#{File.dirname(__FILE__)}/resources/report.html.in"
+    if File.exist?(html_in_path)
+      html_in_path = html_in_path
+    else
+      html_in_path = "#{File.dirname(__FILE__)}/report.html.in"
+    end
+    html_in = ''
+    File.open(html_in_path, 'r') do |file|
+      html_in = file.read
+    end
+
+    # get the weather file run period (as opposed to design day run period)
+    ann_env_pd = nil
+    sql_file.availableEnvPeriods.each do |env_pd|
+      env_type = sql_file.environmentType(env_pd)
+      if env_type.is_initialized
+        if env_type.get == OpenStudio::EnvironmentType.new('WeatherRunPeriod')
+          ann_env_pd = env_pd
+          break
+        end
+      end
+    end
+
+    # only try to get the annual timeseries if an annual simulation was run
+    if ann_env_pd
+
+      # get desired variable
+      key_value = 'Environment'
+      time_step = 'Hourly' # "Zone Timestep", "Hourly", "HVAC System Timestep"
+      variable_name = 'Site Outdoor Air Drybulb Temperature'
+      output_timeseries = sql_file.timeSeries(ann_env_pd, time_step, variable_name, key_value) # key value would go at the end if we used it.
+
+      if output_timeseries.empty?
+        runner.registerWarning('Timeseries not found.')
+      else
+        runner.registerInfo('Found timeseries.')
+      end
+    else
+      runner.registerWarning('No annual environment period found.')
+    end
+
+    raise "I'm purposefully breaking the reporting! Before the report was already created!"
```

Comment:
I raise before the HTML is created in this purposefully_broken_reporting_measure